Source/CronosPlayUnreal/Private/PlayCppSdkBPLibrary.cpp

```diff
@@ -170,11 +171,21 @@ UTexture2D *UPlayCppSdkBPLibrary::GenerateQrCode(FString string) {
       pixels[x + y * size] = color;
     }
   }
+
+#if ENGINE_MAJOR_VERSION == 4
   UTexture2D *texture = UTexture2D::CreateTransient(
       size, size, EPixelFormat::PF_B8G8R8A8, "QRCode");
   void *data = texture->PlatformData->Mips[0].BulkData.Lock(LOCK_READ_WRITE);
   FMemory::Memcpy(data, pixels.GetData(), size * size * 4);
   texture->PlatformData->Mips[0].BulkData.Unlock();
+#else
+  UTexture2D *texture = UTexture2D::CreateTransient(
+      size, size, EPixelFormat::PF_B8G8R8A8, "QRCode");
+  void *data =
+      texture->GetPlatformData()->Mips[0].BulkData.Lock(LOCK_READ_WRITE);
+  FMemory::Memcpy(data, pixels.GetData(), size * size * 4);
+  texture->GetPlatformData()->Mips[0].BulkData.Unlock();
+#endif
```

Comment:
I'm trying to check the difference, and it seems only the last line is different:
```
texture->GetPlatformData()->Mips[0].BulkData.Unlock();
```
vs
```
texture->PlatformData->Mips[0].BulkData.Unlock();
 ```

if that's the case, ifdef can only be on the last line?

Reply:
and this line
```
 void *data =
      texture->GetPlatformData()
   ```
   but line 182  is redundant, i'll fix

Reply:
oh, ok, L182/L176 are duplicate and can be moved up